File: src/1.SRP/domain.dao/EmployeeDAO.ts
import DatabaseConnectionManager from "../detabase/DatabaseConnectionManager";
import Employee from "../domain/Employee";

export default class EmployeeDAO {
    connectionManager = DatabaseConnectionManager.getManagerInstance();

    public saveEmployee(employee: Employee): void {
        this.connectionManager.connect();
        console.log("Employee Record saved")
        this.connectionManager.disconnect();
    }

    public deleteEmployee(employee: Employee): void {
        this.connectionManager.connect();
        console.log("Employee Terminated");
        this.connectionManager.disconnect();
    }
}

File: 1.SRP/src/domain/dao/EmployeeDAO.java
package domain.dao;

import domain.Employee;

import java.sql.SQLException;

public class EmployeeDAO {
    public void saveEmployee(Employee emp) throws SQLException {
//        database.DatabaseConnectionManager connectionManager = database.DatabaseConnectionManager.getManagerInstance();
//        connectionManager.connect();
//        connectionManager.getConnectionObject().prepareStatement("INSERT INTO employee where ");
//        connectionManager.disconnect();
        System.out.println("Saved to DB..");
    }
    public void deleteEmployee(Employee emp) throws SQLException {
//        database.DatabaseConnectionManager connectionManager = database.DatabaseConnectionManager.getManagerInstance();
//        connectionManager.connect();
//        connectionManager.getConnectionObject().prepareStatement("DELETE FROM employee where ");
        System.out.println("Deleted from DB..");
    }
}
